backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/ContractPartnerRepository.java

```diff
@@ -0,0 +1,34 @@
+package de.metas.contracts.bpartner;
+
+import org.adempiere.ad.dao.IQueryBL;
+import org.compiere.model.IQuery;
+import org.compiere.model.I_C_BPartner;
+import org.springframework.stereotype.Repository;
+
+import com.google.common.collect.ImmutableSet;
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.util.Services;
+
+@Repository
+public class ContractPartnerRepository
+{
+	public ImmutableSet<BPartnerId> retrieveAllPartnersWithContracts()
```

Comment:
```suggestion
	public ImmutableSet<BPartnerId> retrieveIdsOfPartnerWithContracts()
```

* include the "Ids" in the name
* in other mf places, "All" implies also inactive, which is not the case here

Reply:
There might be 500k bpartners; please return an iterator
(not sure if i approved the set in a previous PR, but i think it's wrong and we have `IQuery.iterateIds(...)`)

Reply:
Agree. Making it an iterator.